lmod_proxy/static/site/js/site.js

```diff
@@ -0,0 +1,129 @@
+$(document).ready(function(){
+
+  // Setup nice click handling for gradebook
+  $("#gradebook").click(function() {
+    if(!$(this).hasClass("selected")) {
+      $(this).select();
+      $(this).addClass("selected");
+    }
+  });
+  $("#gradebook").blur(function() {
+    if($(this).hasClass("selected")) {
+      $(this).removeClass("selected");
+    }
+  });
+
+
+  function post_call(data) {
+    $('#error').hide(100);
+    console.log(data)
+    $.post(location.pathname, data)
+      .done(function(data, status){
+        console.log(data);
+        console.log(status);
+        $('#result-msg').html(data['msg']);
+        var tbl=$("<table/>").attr("id","datatable");
+        var row = "<thead><tr>";
+        if(typeof data['data'][0] !== 'undefined') {
+          $.each(Object.keys(data['data'][0]), function(index, item) {
```

Comment:
You know, it doesn't seem to be documented that this works in jquery's `each` and `map`. I just assumed it would work from familiarity with how lodash/underscore behaves. I'm not sure how this works under the hood, but I suspect it just changes the behavior based on the number of arguments in the callback passed (all functions have a `length` attribute, but I think this has issues in some browsers and that there are more browser agnostic ways of doing it).

I'd definitely understand not wanting to do this, since it doesn't even seem to be documented: http://api.jquery.com/jQuery.each/ .